packages/blockchain-wallet-v4-frontend/src/data/components/brokerage/slice.ts

```diff
@@ -0,0 +1,115 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import Remote from 'blockchain-wallet-v4/src/remote/remote'
+
+import {
+  AddBankStepType,
+  BankDetailsPayload,
+  BankDWStepType,
+  BrokerageAddBankStepPayload,
+  BrokerageDWStepPayload,
+  BrokerageModalOriginType,
+  BrokerageState,
+  FastLinkType,
+  OBType,
+  YodleeAccountType
+} from './types'
+import { BankTransferAccountType } from 'data/types'
+import { WalletFiatType } from 'core/types'
+import { ModalNamesType } from 'data/modals/types'
+
+const INITIAL_STATE: BrokerageState = {
+  account: undefined,
+  addBankStep: AddBankStepType.ADD_BANK,
+  addNew: false, // TODO: Put this stuff in redux-form
+  bankCredentials: Remote.NotAsked,
+  bankStatus: Remote.NotAsked,
+  bankTransferAccounts: Remote.NotAsked,
+  dwStep: BankDWStepType.DEPOSIT_METHODS,
+  fastLink: Remote.NotAsked,
+  fiatCurrency: undefined,
+  redirectBackToStep: false
+}
+
+const brokerageSlice = createSlice({
+  name: 'brokerage',
+  initialState: INITIAL_STATE,
+  reducers: {
+    createFiatDeposit: () => {},
+    deleteSavedBank: (state, action: PayloadAction<BankTransferAccountType['id']>) => {},
+    handleWithdrawClick: (state, action: PayloadAction<WalletFiatType>) => {},
+    fetchBankLinkCredentials: (state, action: PayloadAction<WalletFiatType>) => {},
+    fetchBankLinkCredentialsLoading: (state) => {
+      state.bankCredentials = Remote.Loading
```

Comment:
These are basically the same reducers we've always had, but notice how we can modify the state directly in here and not run into issues with redux. This is because redux-toolkit uses the immer library behind the scenes to handle creating a new state object for us. 

Now instead of defining a new action and action type in another file, RTK does it for us. At the bottom of this file you can see there is `brokerageSlice.actions` which is all the generated actions such as in this example `brokerageSlice.actions.fetchBankLinkCredentialsLoading` and when we want the action type, you just call `brokerageSlice.actions.fetchBankLinkCredentialsLoading.type` such as in `sagaRegister.ts`